### PR TITLE
adding support of image_diet lib to optimize/compress all static images in collectstatic command + updated with some cleanup.

### PR DESCRIPTION
**Pull request by jacoor**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -15,8 +15,36 @@ System Requirements
 
 Settings
 ========
-``IGNORE_QUNIT_HTML_FILES`` ignore HTML qunits files
-``JSHINT_FILES_FIND`` JS Hint search files grep. Default ``-name "*.js" | xargs grep -l '/\*jslint' | grep -v libs``
+- ``IGNORE_QUNIT_HTML_FILES`` ignore HTML qunits files
+- ``JSHINT_FILES_FIND`` JS Hint search files grep. Default ``-name "*.js" | xargs grep -l '/\*jslint' | grep -v libs``
+- ``STATICFILES_STORAGE = "ydcommon.file_system_storage.YDcommonFileSystemStorage"`` - adds additional compression for all images in staticfiles directories. IMPORTANT: for this to work you need to add 'image_diet' to your INSTALLED APPS
```
Comment (remik) [inline README.rst]: CamelCase would YDCommonFileSystemStorage

File: README.rst [hunk 2]
```diff
@@ -15,8 +15,36 @@ System Requirements
...
+- ``image_diet`` added into installed apps - required by "ydcommon.file_system_storage.YDcommonFileSystemStorage"
+- ``YDCOMMON_COMPRESS_STATIC_IMAGES`` - compress static images when true. Collectstatic takes very long with this, set False for local and dev. Defaults to False.
```
Comment (remik) [inline README.rst]: `True` - uppercase as in Python.

File: ydcommon/file_system_storage.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import os
+from django.contrib.staticfiles.storage import StaticFilesStorage
+from django.conf import settings
+
+
+class YDcommonFileSystemStorage(StaticFilesStorage):
+
+    def post_process(self, files, *args, **kwargs):
+        # print files
```
Comment (remik) [inline ydcommon/file_system_storage.py]: Remove print

File: ydcommon/file_system_storage.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        results = []
+        compress_images = getattr(settings, 'YDCOMMON_COMPRESS_STATIC_IMAGES', False)
+        if 'image_diet' in settings.INSTALLED_APPS and compress_images:
```
Comment (remik) [inline ydcommon/file_system_storage.py]: I would add raise if compress_images but no `image_diet` in apps.

File: README.rst [hunk 5]
```diff
@@ -15,8 +15,36 @@ System Requirements
...
+
+
+Image_diet addons
+========================
+Docs: https://github.com/samastur/image-diet
```
Comment (remik) [inline README.rst]: Please add some summary what it is doing.

Comment (jacoor) [inline README.rst]: Link to doc is not enough?

File: ydcommon/file_system_storage.py (hunk 4 above)
Comment (jacoor) [inline ydcommon/file_system_storage.py]: And why? Image diet is user choice, he may not want to use it. In that case it should not fail.

Comment (jacoor) [inline ydcommon/file_system_storage.py]: Actually rewrote this part.
Raise added.

**Pull request merged**